cmd/core-metadata/res/configuration.toml

```diff
@@ -1,3 +1,8 @@
+# This is required for backwards compatibility so new version of sevice using older 2.x configuration will not fail bootstrapping
+# This will default to false if not provided in old config. Messagebus is now needed by Device System Events and Service Metrics
+# TODO: Remove this setting EdgeX 3.0
+RequireMessageBus = true
+
```

Comment:
Didn't get a chance to review this.

Couldn't we just check if `MessageQueue` or `MessageQueue.protocol` are set, then assume message bus is required?

https://github.com/edgexfoundry/edgex-go/blob/e3ca38a5709b6fd134cd3d7a24fe413a337b80c3/cmd/core-metadata/res/configuration.toml#L64-L65

Reply:
I considered that, but that is ambiguous as to if it is missing because of old configuration or just bad configuration. I prefeed it to be explicate.